pkg/apis/core/types_cloudprofile.go

```diff
@@ -105,6 +105,8 @@ type MachineImageVersion struct {
 	ExpirableVersion
 	// CRI list of supported container runtime and interfaces supported by this version
 	CRI []CRI
+	// Architecture is the CPU architecture of the machine.
+	Architecture string
```

Comment:
New fields added to existing APIs must always be optional, so make this a `*string`

(same for other API types)